src/main/java/org/sonarsource/sonarlint/ls/connected/ProjectBindingManager.java

```diff
@@ -358,6 +366,21 @@ public Map<String, String> getRemoteProjects(@Nullable String maybeConnectionId)
     return Map.of();
   }
 
+  public Map<String, String> getRemoteProjectsByKeys(String connectionId, List<String> projectKeys) {
```

Comment:
Nitpick: this method could itself return a `CompletableFuture` (which could avoid calling `get()` here and re-wrapping it into a `CompletableFuture` to reply to the LSP client).